display-composable-mapview/src/main/java/com/esri/arcgismaps/sample/displaycomposablemapview/MapViewWithCompose.kt

```diff
@@ -0,0 +1,52 @@
+package com.esri.arcgismaps.sample.displaycomposablemapview
+
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.Lifecycle
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.view.MapView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+/**
+ * Wraps the MapView in a Composable function.
+ */
+@Composable
+fun MapViewWithCompose(
+    lifecycle: Lifecycle,
+    arcGISMap: ArcGISMap,
+    viewpoint: Viewpoint,
+    onSingleTap: (mapPoint: Point?) -> Unit = {},
+) {
+    AndroidView(
+        // modifiers are used to set layout parameters
+        modifier = Modifier.fillMaxSize(),
+        // the factory parameter provides a context to create a classic Android view
+        // called when the composable is created, but not when it's recomposed
+        factory = { context ->
+            MapView(context).also { mapView ->
+                // add the MapView to the lifecycle observer
+                lifecycle.addObserver(mapView)
+                // set the map
+                mapView.map = arcGISMap
+                // launch a coroutine to collect map taps
+                CoroutineScope(Dispatchers.Default).launch {
+                    mapView.onSingleTapConfirmed.collect {
+                        onSingleTap(it.mapPoint)
+                    }
+                }
+            }
+        },
+
+        // update block runs every time this view is recomposed which only occurs
+        // when a `State<T>` or `MutableState<T>` parameter is changed.
+        update = { view -> // view is automatically cast to a MapView
+            view.setViewpoint(viewpoint) // called only if the viewpoint is changed
+        }
```

Comment:
I think if you are going to have the map as a parameter, then you should also update it here. Otherwise you are breaking the contract because a user might call this compose function again with a different map and as written, the map would not update. So I would add this line or remove the map parameter
```suggestion
        update = { view -> // view is automatically cast to a MapView
            view.setViewpoint(viewpoint) // called only if the viewpoint argument is changed
            view.map = arcGISMap // called only if the arcGISMap argument is changed
        }
```

Reply:
My understanding is that any MapView property that is modified after initial composition would need to be reflected in the `update` block. Since the sample does not modify the map, it's not used in the `update` block. This is why `ArcGISMap` parameter's state is not stored by `remember` during initial composition. The only mutable state stored is the `ViewPoint` which is used in the `update` block. If my understanding is right here, this sample shouldn't be breaking any contract. 😅 


Do you think it would be helpful to showcase both `ArcGISMap` and `Viewpoint` as mutable states even if ArcGISMap is not updated? I feel like the user will have to take this into account for any MapView property that would be modified after initial composition like for example adding adding a new graphic to `graphicsOverlays` property.

Reply:
For me I think the contract is that a parameter to a compose function should be expected to change. What happens if the user _does_ use a stateful map argument that changes after the MapView is initialized? For this sample it's not that important, I just feel that if the map is a parameter then it should also be in the update block, unless it is called eg. `initialMap`, because then it's clear that the parameter is only for the first setup. For me I would either make the map a mutable state or remove the parameter and just hardcode the map that gets set in the factory bock.

Reply:
> For me I think the contract is that a parameter to a compose function should be expected to change.

Agreed, this makes more sense 👍.  I've updated the code to use a stateful map which is accessed in `update`.